test/common/common/stack_array_test.cc

```diff
@@ -0,0 +1,38 @@
+#include "common/common/stack_array.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+class TestEntry {
+public:
+  TestEntry() { constructed_ = true; }
+  ~TestEntry() { destructor_(val_); }
+
+  int val_ = 0;
+  bool constructed_ = false;
```

Comment:
this seems strange as there won't be a time when the test-method has control where constructed_ will be false.

I thought for a couple of minutes how to gain confidence your ctor got run. What you are doing below would also work if StackArray failed to construct anything, but the stack-memory had random garbage in it (which would be interpreted as truthy.

Maybe have a member-var TestEntry* self_, and init it in the ctor to this.  Then you can check that happened.